uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestRequired_uefi.c

```diff
@@ -714,6 +732,101 @@ CheckRuntimeServices (
   return EFI_SUCCESS;
 }
 
+EFI_STATUS
+CheckRuntimePropertiesTable (
+  IN EFI_STANDARD_TEST_LIBRARY_PROTOCOL   *StandardLib
+  )
+{
+  EFI_TEST_ASSERTION       AssertionType;
+  EFI_RT_PROPERTIES_TABLE  *RtPropertiesTable               = NULL;
+  UINT32                   ExpectedRuntimeServicesSupported = 0u;
+  EFI_STATUS               Status                           = EFI_NOT_STARTED;
+
+  RUNTIME_SERVICE_CHECK    RuntimeServices[] = {
+    { (VOID *)gtRT->GetTime, EFI_RT_SUPPORTED_GET_TIME },
+    { (VOID *)gtRT->SetTime, EFI_RT_SUPPORTED_SET_TIME },
+    { (VOID *)gtRT->GetWakeupTime, EFI_RT_SUPPORTED_GET_WAKEUP_TIME },
+    { (VOID *)gtRT->SetWakeupTime, EFI_RT_SUPPORTED_SET_WAKEUP_TIME },
+    { (VOID *)gtRT->GetVariable, EFI_RT_SUPPORTED_GET_VARIABLE },
+    { (VOID *)gtRT->GetNextVariableName, EFI_RT_SUPPORTED_GET_NEXT_VARIABLE_NAME },
+    { (VOID *)gtRT->SetVariable, EFI_RT_SUPPORTED_SET_VARIABLE },
+    { (VOID *)gtRT->SetVirtualAddressMap, EFI_RT_SUPPORTED_SET_VIRTUAL_ADDRESS_MAP },
+    { (VOID *)gtRT->ConvertPointer, EFI_RT_SUPPORTED_CONVERT_POINTER },
+    { (VOID *)gtRT->GetNextHighMonotonicCount, EFI_RT_SUPPORTED_GET_NEXT_HIGH_MONOTONIC_COUNT },
+    { (VOID *)gtRT->ResetSystem, EFI_RT_SUPPORTED_RESET_SYSTEM },
+    { (VOID *)gtRT->UpdateCapsule, EFI_RT_SUPPORTED_UPDATE_CAPSULE },
+    { (VOID *)gtRT->QueryCapsuleCapabilities, EFI_RT_SUPPORTED_QUERY_CAPSULE_CAPABILITIES },
+    { (VOID *)gtRT->QueryVariableInfo, EFI_RT_SUPPORTED_QUERY_VARIABLE_INFO }
+  };
+
+  //
+  // Check the EFI Runtime Services Table
+  //
+  Status = EfiGetSystemConfigurationTable (&gEfiRtPropertiesTableGuid, (VOID **)&RtPropertiesTable);
+  if (EFI_ERROR (Status)) {
+    AssertionType = EFI_TEST_ASSERTION_FAILED;
+  } else {
+    AssertionType = EFI_TEST_ASSERTION_PASSED;
+  }
+
+  StandardLib->RecordAssertion (
+                 StandardLib,
+                 AssertionType,
+                 gEfiCompliantBbTestRequiredAssertionGuid004,

```

Comment:
Do these assertion guids have specific meaning?

Reply:
I didn't find anything within the source code that explained these GUIDS. This is the only comment that I read, 

// GUIDs auto-generated for EFI test assertion.

So, I didn't want to create any new GUIDS and I used the GUIDs in the same situations/scenarios as the rest of the code.

Reply:
Hm okay. I wonder if the GUIDs need to be unique. Is there any issue when you run the `CheckLoadedImageProtocol` test before/after the `CheckRuntimePropertiesTable` test? That one is using the same assertion GUID.

Reply:
There is no actual issue if we use the same GUID, I am currently running all of the tests under "Generic" tests. The GUID is  used for logging the tests GUID in the logs as shown below after every Result.  However, it is possible that they expect for every entry in the log to have a unique GUID.  I did notice though,  that each method will use the same GUID if there are multiple RecordAssertions within the method. That may be a way to group multiple result entries to the same test GUID. 

I have two RecordAssertions in my method, so I believe it would be correct to use the same GUID on both of my RecordAssertions.   What I have concluded from the code is that each method should use a unique GUID, and if a method has multiple RecordAssertions, they should use the same GUID. 

Log Snippet:
> 
UEFI Compliant - EFI Runtime Properties Table RuntimeServicesSupported variable must be implemented -- PASS
13A20958-C860-452F-B9A2-E6D996419224
f:\building_uefi_sct\edk2\SctPkg\TestCase\UEFI\EFI\Generic\EfiCompliant\BlackBoxTest\EfiCompliantBBTestRequired_uefi.c:861:RuntimeServicesSupported - 0x3FFF, Expected - 0x3FFF

UEFI Compliant - LOADED_IMAGE protocol must exist -- PASS
A82F8D56-1476-41F1-BAC4-9759799F97F3
f:\building_uefi_sct\edk2\SctPkg\TestCase\UEFI\EFI\Generic\EfiCompliant\BlackBoxTest\EfiCompliantBBTestRequired_uefi.c:902:Status - Success, Expected - Success
>

Reply:
In order to follow the same pattern, it seems I would need to create a new GUID for the test, because there currently doesn't exist an Assertion 10 and that is the next available GUID that I should use to keep the GUIDs unique.  

#define EFI_TEST_EFICOMPLIANTBBTESTREQUIRED_ASSERTION_010_GUID \

I think I will auto generate a new GUID using the Visual Studio GUID generator and use the new GUID 10 for the new test.

Reply:
Here is the log after creating a new GUID. 
[RequiredElements_0_0_117C9ABC-489D-4504-ACDB-12AACE8F505B.log](https://github.com/user-attachments/files/17851833/RequiredElements_0_0_117C9ABC-489D-4504-ACDB-12AACE8F505B.log)

Reply:
Yes, I think it's safe to use a new GUID to group this test. The log you attached is reporting a failure and there's also mentions of UEFI 2.6. Is this expected?

Reply:
Yes, this log was for a failed case. The system I am using for a passed case is the Warpath and I will upload that log as well. 

Regarding the UEFI 2.6, that is the global flag used throughout the project, 

#define BIOS_ID_DEFAULT                     L"UEFI 2.6"

But I felt I should hold off on that, because this SCT is at version 2.7B and they have yet to update this value.

Reply:
Makes sense. I agree we shouldn't modify the version. I was just wondering whether the test are grouped by spec version  or whether the entire test is just reporting a single version.

Reply:
[RT_Properties_Passed_New_GUID.log](https://github.com/user-attachments/files/17877348/RT_Properties_Passed_New_GUID.log)

Reply:
[RT_Properties_Passed_New_GUID.log](https://github.com/user-attachments/files/17877362/RT_Properties_Passed_New_GUID.log)